api/v1/intrusiondetection_types.go

```diff
@@ -17,11 +17,16 @@ limitations under the License.
 package v1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // IntrusionDetectionSpec defines the desired state of Tigera intrusion detection capabilities.
 type IntrusionDetectionSpec struct {
+	// ComponentResources can be used to customize the resource requirements for each component.
+	// Only DeepPacketInspection is supported for this spec.
+	// +optional
+	ComponentResources []IntrusionDetectionComponentResource `json:"componentResources,omitempty"`
```

Comment:
Was this structure agreed on already?

I don't think this should be an arbitrary array of resources, there are going to be a finite number of intrusion detection components, and not many of them, I would we would just want specific component fields, maybe nested under the Resource field.

Reply:
Yes, this was the agreed structure. It is not a new structure, it is already used by installation and logstorage component to set resource requirements
https://github.com/tigera/operator/blob/271d1b851db73b758c6c986f7f2f5170e253b724/api/v1/logstorage_types.go#L54
https://github.com/tigera/operator/blob/271d1b851db73b758c6c986f7f2f5170e253b724/api/v1/installation_types.go#L132

Reply:
Ah OK great.